components/friend.php

```diff
@@ -69,18 +59,25 @@ public function create( $args, $assoc_args ) {
 			WP_CLI::error( 'No user found by that username or ID.' );
 		}
 
-		if ( ! friends_add_friend( $initiator->ID, $friend->ID, $r['force-accept'] ) ) {
+		$force = \WP_CLI\Utils\get_flag_value( $assoc_args, 'force-accept' );
+
+		// Check if already friends, and bail if so.
+		if ( friends_check_friendship( $initiator->ID, $friend->ID ) ) {
+			WP_CLI::error( 'These users are already friends.' );
+		}
+
+		if ( ! friends_add_friend( $initiator->ID, $friend->ID, $force ) ) {
 			WP_CLI::error( 'There was a problem while creating the friendship.' );
 		}
 
-		if ( $r['silent'] ) {
+		if ( \WP_CLI\Utils\get_flag_value( $assoc_args, 'silent' ) ) {
```

Comment:
Missed this during the initial review. If the creation errors, the process will stop, even if `--silent` - see two lines up. I think this is probably OK, but we should be consistent throughout wp-cli-buddypress.

Reply:
@boonebgorges I didn't catch what you meant. If the process has an error, it is true that `--silent` will not be called.

Is it that correct?! Makes sense to me.

Reply:
I mean: If the friendship creation fails, then `WP_CLI::error()` will cause execution to be stopped. So `friend generate --count=5` could stop after 2 friendships if the third one fails.

Reply:
That's true! I guess I have to update in other places, in the package, to avoid this error.

Thank you for bringing this to my attention. :)